src/Lockedfield.php

```diff
@@ -210,19 +210,26 @@ public function getLocks($itemtype, $items_id, bool $fields_only = true)
         /** @var \DBmysql $DB */
         global $DB;
 
-        $iterator = $DB->request([
+        $subquery1 = new \QuerySubQuery([
             'FROM'   => $this->getTable(),
             'WHERE'  => [
                 'itemtype'  => $itemtype,
-                [
-                    'OR' => [
-                        'items_id'  => $items_id,
-                        'is_global' => 1
-                    ]
-                ]
+                'items_id'  => $items_id,
             ]
         ]);
 
+        $subquery2 = new \QuerySubQuery([
+            'FROM'   => $this->getTable(),
+            'WHERE'  => [
+                'is_global' => 1
```

Comment:
There is a missing criteria here.

```suggestion
                'itemtype' => $itemtype,
                'is_global' => 1,
```